proto/outbound.proto

```diff
@@ -436,3 +444,15 @@ message ExponentialBackoff {
   // Must be greater than or equal to 0.0.
   float jitter_ratio = 3;
 }
+
+// Error type that is used to indicate that any traffic
+// that is delivered through a route should be failed.
+message RouteError {
+  string message = 1;
+}
```

Comment:
Rather than use free-form string messages to indicate denials, I suspect we want a oneof or enum to clearly enumerate the types of route errors that may be emitted.

I'd also be inclined to move this message within the Route messages, as we do for other types.